backend/core/src/primary-adapters/rest/history/history-controller.ts

```diff
@@ -0,0 +1,108 @@
+import { BaseHttpController, controller, httpGet, httpPost, requestBody } from 'inversify-express-utils';
+import { inject } from 'inversify';
+import { CONTAINER_TYPES, HistoryRequestDto, HistoryResponseDto } from '~/shared/types/types';
+import { HistoryService } from '~/core/history/application/history-service';
+import { History } from '@prisma/client';
+import { authenticationMiddleware } from '../middleware';
+
+/**
+ * @swagger
+ * tags:
+ *   name: history
+ *   description: History management
+ * components:
+ *    schemas:
+ *      History:
+ *        type: object
+ *        properties:
+ *          id:
+ *            type: string
+ *            format: uuid
+ *          userId:
+ *            type: string
+ *            format: uuid
+ *          videoId:
+ *            type: string
+ *            format: uuid
+ *          createdAt:
+ *            type: string
+ *            format: date-time
+ */
+@controller('/history')
+export class HistoryController extends BaseHttpController {
+  private historyService: HistoryService;
+
+  constructor(@inject(CONTAINER_TYPES.HistoryService) historyService: HistoryService) {
+    super();
+
+    this.historyService = historyService;
+  }
+
+  /**
+   * @swagger
+   * /history:
+   *    get:
+   *      tags:
+   *      - history
+   *      operationId: getAllUserHistory
+   *      description: Returns an array of user history
+   *      security:
+   *        - bearerAuth: []
+   *      responses:
+   *        200:
+   *          description: Successful operation
+   *          content:
+   *            application/json:
+   *              schema:
+   *                type: array
+   *                items:
+   *                  $ref: '#/components/schemas/History'
+   *        401:
+   *          $ref: '#/components/responses/NotFound'
+   */
+  @httpGet('/', authenticationMiddleware)
+  public getAllUserHistory(@requestBody() historyRequestDto: { userId: string }): Promise<History[]> {
+    const { userId } = historyRequestDto;
+    return this.historyService.getAllUserHistory(userId);
+  }
```

Comment:
Any authorized user can get another user's story

Reply:
No problem, put another userId in a body request.

Reply:
Done!